File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.4. PAO202650_RESTFUL_DOTNET_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_DOTNET_GR5/CONUNI_RESTFUL_DOTNET_GR9S/Models/ConversionRequest.cs
namespace CONUNI_RESTFUL_DOTNET_GR9S.Models
{
    /// <summary>
    /// Modelo para las peticiones de conversión
    /// </summary>
    public class ConversionRequest
    {
        public double Valor { get; set; }
        public string UnidadOrigen { get; set; }
        public string UnidadDestino { get; set; }
        public string Categoria { get; set; }
        
        public ConversionRequest()
        {
        }
        
        public ConversionRequest(double valor, string unidadOrigen, string unidadDestino, string categoria)
        {
            Valor = valor;
            UnidadOrigen = unidadOrigen;
            UnidadDestino = unidadDestino;
            Categoria = categoria;
        }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.3. PAO202650_RESTFUL_JAVA_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_JAVA_GR05/src/main/java/ec/edu/monster/modelo/ConversionRequest.java
package ec.edu.monster.modelo;

import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Modelo para las peticiones de conversión
 * @author ACER NITRO V15
 */
@XmlRootElement
public class ConversionRequest {
    
    private double valor;
    private String unidadOrigen;
    private String unidadDestino;
    private String categoria;
    
    // Constructores
    public ConversionRequest() {
    }
    
    public ConversionRequest(double valor, String unidadOrigen, String unidadDestino, String categoria) {
        this.valor = valor;
        this.unidadOrigen = unidadOrigen;
        this.unidadDestino = unidadDestino;
        this.categoria = categoria;
    }
    
    // Getters y Setters
    public double getValor() {
        return valor;
    }
    
    public void setValor(double valor) {
        this.valor = valor;
    }
    
    public String getUnidadOrigen() {
        return unidadOrigen;
    }
    
    public void setUnidadOrigen(String unidadOrigen) {
        this.unidadOrigen = unidadOrigen;
    }
    
    public String getUnidadDestino() {
        return unidadDestino;
    }
    
    public void setUnidadDestino(String unidadDestino) {
        this.unidadDestino = unidadDestino;
    }
    
    public String getCategoria() {
        return categoria;
    }
    
    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }
    
    @Override
    public String toString() {
        return "ConversionRequest{" +
                "valor=" + valor +
                ", unidadOrigen='" + unidadOrigen + '\'' +
                ", unidadDestino='" + unidadDestino + '\'' +
                ", categoria='" + categoria + '\'' +
                '}';
    }
}


